homeassistant/components/opensprinkler/sensor.py

```diff
@@ -0,0 +1,147 @@
+"""Opensprinkler integration."""
+from datetime import datetime
+import logging
+from typing import Callable
+
+import pytz
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from . import OpensprinklerSensor
+from .const import DATA_DEVICES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+UTC_TZ = pytz.timezone("UTC")
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: dict,
+    async_add_entities: Callable,
+    discovery_info: dict,
+):
+    """Set up the opensprinkler sensors."""
+    entities = await hass.async_add_executor_job(
+        _create_entities, hass, config, discovery_info
+    )
+    async_add_entities(entities)
+
+
+def _create_entities(hass: HomeAssistant, config: dict, discovery_info: dict):
```

Comment:
What code is doing I/O in this function? 

If it's `device.stations` and you also maintain the library, I would suggest to rewrite properties that do I/O to functions. It's very difficult to see what is doing I/O otherwise.